src/GraphQL/ResolveFieldContext/ReadonlyResolveFieldContext.cs

```diff
@@ -43,10 +43,10 @@ internal ReadonlyResolveFieldContext Reset(ExecutionNode? node, ExecutionContext
         }
 
         private IDictionary<string, ArgumentValue>? GetArguments()
-            => ExecutionHelper.GetArguments(_executionNode.FieldDefinition!.Arguments, _executionNode.Field!.Arguments, _executionContext.Variables);
+            => _executionContext.ArgumentValues?.TryGetValue(FieldAst, out var ret) ?? false ? ret : FieldDefinition.DefaultArguments;
```

Comment:
`_arguments` and _directives` fields are not needed anymore. The same for both private methods.